xonsh/built_ins.py

```diff
@@ -885,10 +885,11 @@ def list_of_strs_or_callables(x):
     """Ensures that x is a list of strings or functions"""
     if isinstance(x, str) or callable(x):
         rtn = [x]
-    elif isinstance(x, cabc.Sequence):
-        rtn = [i if isinstance(i, str) or callable(i) else str(i) for i in x]
     else:
-        rtn = [str(x)]
+        try:
+            rtn = [i if isinstance(i, str) or callable(i) else str(i) for i in x]
+        except:
```

Comment:
I'd really prefer not to use try-except blocks for things that are deterministically knowable.  Also, we don't allow bare excepts, so at the very least, this should be `except Exception`.